platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/RosterUtils.java

```diff
@@ -49,7 +70,117 @@ public static String formatNodeName(final long nodeId) {
      * @return its Hash
      */
     @NonNull
-    public static Hash hash(@NonNull final Roster roster) {
+    public static Hash hashOf(@NonNull final Roster roster) {
         return PBJ_RECORD_HASHER.hash(roster, Roster.PROTOBUF);
     }
+
+    /**
+     * Constructs the initial roster based on the given software version and initial state.
+     * The initial roster is constructed by adopting the candidate roster if a software upgrade is detected.
+     * Otherwise, the initial roster is created from the address book.
+     *
+     * @param version      the software version
```

Comment:
Which software version? The version of the current software running or the version of the state?

Reply:
It's the currently running software version. Updated.